lib/resty/core/request.lua

```diff
@@ -346,6 +349,25 @@ function ngx.req.clear_header(name)
 end
 
 
+function ngx.req.get_uri_exten()
+    local r = getfenv(0).__ngx_req
+    if not r then
+        return error("no request found")
+    end
+
+    local buf = get_string_buf(255)
+    local sizep = get_size_ptr()
+    sizep[0] = 255
```

Comment:
We should avoid using magical number like 255 directly in the code. Give it a name. It will also make it easier to update the number in the future.